sdk/src/main/scala/io/horizen/account/forger/AccountForgeMessageBuilder.scala

```diff
@@ -6,7 +6,7 @@ import io.horizen.account.block.AccountBlock.calculateReceiptRoot
 import io.horizen.account.block.{AccountBlock, AccountBlockHeader}
 import io.horizen.account.chain.AccountFeePaymentsInfo
 import io.horizen.account.companion.SidechainAccountTransactionsCompanion
-import io.horizen.account.fork.{Version1_2_0Fork, GasFeeFork}
+import io.horizen.account.fork.{GasFeeFork, Version1_2_0Fork, Version1_4_0Fork}
 import io.horizen.account.history.AccountHistory
 import io.horizen.account.mempool.{AccountMemoryPool, MempoolMap, TransactionsByPriceAndNonceIter}
 import io.horizen.account.proposition.AddressProposition
```

Comment:
In getForgingStakeMerklePathInfo method, at line 412
```
       stateViewFromRoot =>
        stateViewFromRoot.getOrderedForgingStakesInfoSeq(nextConsensusEpochNumber)
```
shouldn't be used nextConsensusEpochNumber - 2?

Reply:
True, the problem is that for calculating the percentages of stakes used by the lottery  it is used a totalStake value that was calculated without the filtering and so taking into account all the forgers, not only the ones above minimumStake. This could cause that, only for the 2 epochs after the activation of fork 1.4, fewer blocks will be forged.
In AbstractForgeMessageBuilder.tryToForgeNextBlock, totalStake, used in vrfProofCheckAgainstStake for calculating the percentages, is retrieved from the history and it was also saved 2 epochs before but without the filtering, because the fork wasn't active yet.
I'll make an example for better explain the issue.
Let's say the fork 1.4 is activated at epoch 40.
At epoch 38, the stakes are:
- Forger 1 20 stakes
- Forger 2 15 stakes
- Forger 3 8 stakes
for a totalStake of 43.
At the end of epoch 38, in AbstractSidechainNodeViewHolder.scala , in applyStateAndWallet between lines 400-411, the totalStake is saved in the history with a value of 43, because the fork 1.4 is not active yet.
At the epoch 40, the fork is activated. When a block is forged, the stakes related to epoch 38 are retrieved, including the totalStake. Because the block belongs to epoch 40, only forger 1 and 2 will be selected for the lottery, but their percentages will be lower that they should be because it will be used 43 instead of 35 as totalStake. So there will be a lower chance to forge a block.
This problem exists only for 2 epochs after the fork, after that everything will be consistent because the totalStake will be calculated filtering the forgers without enough stakes.

Reply:
yes agree with @saratnt 
as I told on slack the epoch passed to getOrderedForgingStakesInfoSeq is only used to verify the fork activatyion, so we need to pass the epoch for which we are extracting the stakes (nextConsensusEpochNumber - 2)

Reply:
Thanks for the explanation. Added stakeEpochNumber input param to the getOrderedForgingStakesInfoSeq.
It is epoch - 2 when called from forgeMessageBuilder, and just epoch when called from nodeViewHolder